text/0000-annotations.md

```diff
@@ -0,0 +1,78 @@
+# Meta
+[meta]: #meta
+- Name: OCI Image annotations
+- Start Date: 2021-12-03
+- Author(s): [@samj1912](https://github.com/samj1912)
+- Status: Draft <!-- Acceptable values: Draft, Approved, On Hold, Superseded -->
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: (put "N/A" unless this replaces an existing RFC, then link to that RFC)
+
+# Summary
+[summary]: #summary
+
+Add OCI recommended image annotations to output images.
+
+# Definitions
+[definitions]: #definitions
+
+N/A
+
+# Motivation
+[motivation]: #motivation
+
+Certain OCI image scanning and debugging tools take into account standard annotations on the images as defined by [OCI Spec - Pre-defined Annotation keys](https://github.com/opencontainers/image-spec/blob/main/annotations.md#pre-defined-annotation-keys). Buildpacks already records some of this information in the `project.toml` and other labels defined at https://github.com/buildpacks/spec/blob/main/platform.md#labels. We should also preserve this information in the output image in appropriate annotations as recommended by the OCI spec.
+
+# What it is
+[what-it-is]: #what-it-is
+
+Adds OCI recommended annotations to the output image.
+
+# How it Works
+[how-it-works]: #how-it-works
+
+Optionally adds the following annotations where applicable when lifecycle is used to publish images directly to the registry - 
+
+- `org.opencontainers.image.base.name`: Run image tag name. Currently stored in `io.buildpacks.lifecycle.metadata.run-image.reference`
+- `org.opencontainers.image.base.digest`: Run image digest. Currently stored in `io.buildpacks.lifecycle.metadata.run-image.reference`
+
+The platform may also provide appropriate keys inside `project-metadata.toml` to the lifecycle so that it exports the appropriate annotations listed below.
```

Comment:
In the ideal world I think these project metadata annotation would be better contributed by a buildpack vs by the lifecycle for the following reasons:
1. Buildpack contributed behavior more easily upgradable accross a variable of platforms
2. Keeping the surface area of the platform API as small as possible makes the spec and the conditional logic in lifecycle less complicated, thus making our lives easier and enabling us to actually make progress shipping things.

The way I imagine it:
* **step 1** remove daemon
* **step 2** allow buildpacks to contribute aribtrary annotations
* **step 3** provide a (utility?) buildpack that contributes these annotations based off the values in `project.toml`

Reply:
That ordering makes sense to me. Is there a rough plan/timeline for step 1?

Reply:
It ties into this RFC issue: https://github.com/buildpacks/rfcs/issues/197

@jjbustamante is on it!

Reply:
Based on my comment [here](https://github.com/buildpacks/rfcs/pull/196#discussion_r789017080), I'm back to say that I don't think step 1 is necessary. 

steps 2 and 3 can be an alternative approach to this RFC without step 1. I'd love to see it comparatively in the `Alternatives` seciton.

Reply:
https://github.com/buildpacks/rfcs/pull/196#discussion_r789142013